packages/core/src/events/augury-event.class.ts

```diff
@@ -1,29 +1,41 @@
-import { AuguryEventName } from './augury-event-name.type';
+import { Probe } from '../probes';
+import { DragPeriod } from './drag-period.class';
 
 export class AuguryEvent {
   private static NextId = 0;
 
   public readonly id: number;
-  public readonly payload: any;
-  public readonly creationAtTimestamp: number;
-  public readonly completedAtTimestamp: number;
-
-  constructor(
-    public readonly probeName: string,
-    public readonly name: AuguryEventName,
-    createPayload: () => any,
-  ) {
+  public readonly probeName: string;
+  private timestamps: DragPeriod = null;
+
+  constructor(probe: Probe) {
```

Comment:
Feels a bit funny couple the event to a probe in this way. Since we have the event type to we need a name identifier as well? Seems to me this is more or less used as a category or tag in how it's utilized.

Reply:
I think it might actually be useful to say, "I'm interested in analyzing all events coming from this probe", assuming there's more than one. You could also just instanceOf a big list of event types too and achieve the same thing I guess. I mostly just wanted to prevent repeating probe.constructor.name over and over and over when creating events.